reference/preconditioner/batch_isai_kernels.hpp.inc

```diff
@@ -92,11 +87,38 @@ inline void extract_pattern_for_dense_sys_corr_to_current_row_impl(
         const int A_row_st = A_row_ptrs[A_row];
         const int A_row_end = A_row_ptrs[A_row + 1];
 
-        // match row: curr_row_idx of aiA with row: A_row of A (both the arrays
-        // are in sorted order)
-        match_and_write(size, i - aiA_row_st, aiA_row_st, aiA_row_end,
-                        aiA_col_idxs, A_row_st, A_row_end, A_col_idxs,
-                        dense_mat_pattern_ptr);
+
+        if (size <= row_size_limit) {
```

Comment:
Do you need a special case for `size <= row_size_limit` ? Can it not be unified with the general case ?

Reply:
-  From the cuda backend point of view, the special case has performance advantages.

- Coming to the reference kernels, yes both the cases could be clubbed, but that would require changes in the core part. Currently, the check to see whether the general case needs to be executed is located in the core. Unifying both the cases for reference would mean moving this check to individual backends and having different implementations for it. It could be done but the downside is that sometimes the iterative solver used in the general case does not converge, so the reference isai output could be skewed even in cases (all rows: nnz_per_row <= 32) where a clear-cut solution is possible.

Reply:
I see. Then lets leave it as it is.